parser_library/src/processing/instruction_sets/asm_processor.cpp

```diff
@@ -130,7 +155,7 @@ void asm_processor::process_EQU(rebuilt_statement stmt)
 
         if (expr_op && !expr_op->has_dependencies(dep_solver))
         {
-            auto length_value = expr_op->expression->resolve(dep_solver);
+            auto length_value = expr_op->expression->evaluate(dep_solver, *this);
```

Comment:
It feels a little bit weird to provide `*this` for the `diagnostic_op_consumer...` in call to `evaluate(...)`.

Is it possible/meaningful to not have this inheritance relationship on `diagnosable_ctx` and have it in a form of a member variable instead?

Reply:
eventually yes, but the PR seems quite big already, so I thought leaving it for another PR in the future... at least there is only handful of processors unlike thousands of expressions.